File: algorithms/word_count/word_count.py
def word_count(s):
    counts = {}
    to_ignore = '" : ; , . - + = / \ | [ ] { } ( ) * ^ &'

    s = s.lower()

    string_list = s.split()

    for word in string_list:
        # strip out invalid chars
        word = word.strip(to_ignore)

        # If there are no valid chars, skip to the next word
        if word == '': continue

        if word in counts:
            counts[word] += 1
        else:
            counts[word] = 1

    return counts

File: algorithms/word_count/wordCount.ts
export const wordCount = (s: string): object => {
  const counts = {};

  // Regex expression to select all characters
  // that are not ' or alphabetical
  const regex = /[^a-zA-Z']/g;

  s = s.replace(regex, " ");

  s = s.toLowerCase();

  const wordList = s.split(" ");

  wordList.forEach((word) => {
    if (word.length === 0) {
      return;
    }

    const keys = Object.keys(counts);

    if (keys.includes(word)) {
      counts[word]++;
    } else {
      counts[word] = 1;
    }
  });

  return counts;
};
